annotation-processor/src/main/java/org/hibernate/validator/ap/classchecks/AbstractMethodOverrideCheck.java

```diff
@@ -88,105 +79,88 @@ public AbstractMethodOverrideCheck(Elements elementUtils, Types typeUtils, Const
 			return;
 		}
 
-		ExecutableElement element = getOverriddenElement( currentTypeElement, superType, currentMethod );
-		if ( element != null ) {
-			tree.addNode( element, superType, currentTypeElement );
-			findAllOverriddenElementsRecursive( superType, element, tree );
+		ExecutableElement overriddenMethod = getOverriddenMethod( overridingMethod, superclassTypeElement );
+		if ( overriddenMethod != null ) {
+			methodInheritanceTreeBuilder.addOverriddenMethod( overridingMethod, overriddenMethod );
+			overridingMethod = overriddenMethod;
 		}
+
+		collectOverriddenMethods( overridingMethod, superclassTypeElement, methodInheritanceTreeBuilder );
 	}
 
 	/**
-	 * Find pairs of enclosing type {@link TypeElement} and overridden method {@link ExecutableElement}  from implemented interfaces.
-	 *
-	 * @param currentTypeElement the class in which the method is located
-	 * @param interfaces a list of implemented interfaces
-	 * @param currentMethod the method for which we want to find the overridden methods
+	 * Collect overridden methods in the interfaces of a given type.
 	 *
-	 * @return a map of pairs of overridden methods (map key - an enclosing type, map value - overridden method in that type)
-	 * if there are any, an empty map otherwise
+	 * @param overridingMethod the method for which we want to find the overridden methods
+	 * @param currentTypeElement the class we are currently analyzing
+	 * @param methodInheritanceTreeBuilder the method inheritance tree builder
 	 */
-	private Map<TypeElement, ExecutableElement> findOverriddenMethodInInterfacesPairs(
-			TypeElement currentTypeElement,
-			List<? extends TypeMirror> interfaces,
-			ExecutableElement currentMethod) {
-		Map<TypeElement,ExecutableElement> elements = CollectionHelper.newHashMap();
-
-		for ( TypeMirror anInterface : interfaces ) {
-			TypeElement implementedInterface = (TypeElement) typeUtils.asElement( anInterface );
-			ExecutableElement element = getOverriddenElement( currentTypeElement, implementedInterface, currentMethod );
-			if ( element != null ) {
-				elements.put( implementedInterface, element );
+	private void collectOverriddenMethodsInInterfaces(ExecutableElement overridingMethod, TypeElement currentTypeElement,
+			MethodInheritanceTree.Builder methodInheritanceTreeBuilder) {
+		for ( TypeMirror implementedInterface : currentTypeElement.getInterfaces() ) {
+			TypeElement interfaceTypeElement = (TypeElement) typeUtils.asElement( implementedInterface );
+			ExecutableElement overriddenMethod = getOverriddenMethod( overridingMethod, interfaceTypeElement );
+			if ( overriddenMethod != null ) {
+				methodInheritanceTreeBuilder.addOverriddenMethod( overridingMethod, overriddenMethod );
+				collectOverriddenMethodsInInterfaces( overriddenMethod, interfaceTypeElement, methodInheritanceTreeBuilder );
+			}
+			else {
+				collectOverriddenMethodsInInterfaces( overridingMethod, interfaceTypeElement, methodInheritanceTreeBuilder );
```

Comment:
There's a slight difference in the calls: one is using `overriddenMethod` and the other `overridingMethod`. I thought it was clearer this way than to declare another variable. Your question makes me think I might be wrong. WDYT?